### PR TITLE
Added e2e cypress tests

### PR DESCRIPTION
**Pull request by MyNewGitHere**
Added End 2 End tests using Cypress

### CODE REVIEW

File: cypress/e2e/homepage.cy.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    cy.intercept(
+      "GET",
+      "https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty"
+    ).as("getTopStories");
+    cy.get('[data-testid="new-button"]').click();
+    cy.get('[data-testid="past-button"]').click();
+    cy.wait("@getTopStories").then((interception) => {
+      if (interception.response) {
+        expect(interception.response.statusCode).to.eq(200);
+      }
```
Comment (aashitk) [inline cypress/e2e/homepage.cy.ts]: Why we need if condition here?

File: cypress/e2e/homepage.cy.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      expect(interception.request.url).to.eq(
+        "https://hacker-news.firebaseio.com/v0/newstories.json?print=pretty"
+      );
+      cy.get('[data-testid="new-button"]').should(
+        "have.css",
+        "background-color",
+        "rgb(251, 201, 27)"
+      );
+    });
+  });
```
Comment (aashitk) [inline cypress/e2e/homepage.cy.ts]: Check assertion of news-cards as well.

File: cypress/e2e/homepage.cy.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      expect(interception.request.url).to.eq(
+        "https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty"
+      );
+      cy.get('[data-testid="past-button"]').should(
+        "have.css",
+        "background-color",
+        "rgb(251, 201, 27)"
+      );
+    });
+  });
```
Comment (aashitk) [inline cypress/e2e/homepage.cy.ts]: Check assertion of news-cards as well.

File: cypress/e2e/homepage.cy.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  it("VerifyLoadMoreButton", () => {
+    cy.get('[data-testid="load-more-button"]').click();
+    cy.get('[data-testid="news-card"]').should("have.length", 10);
+  });
+
+  it("VerifyNewsCardNavigation", () => {
+    cy.get('[data-testid="news-card"]').first().click();
```
Comment (aashitk) [inline cypress/e2e/homepage.cy.ts]: This it block is missing assertion to check navigation.

File: cypress/e2e/homepage.cy.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        "have.css",
+        "background-color",
+        "rgb(251, 201, 27)"
+      );
+    });
+  });
+
+  it("VerifyLoadMoreButton", () => {
+    cy.get('[data-testid="load-more-button"]').click();
+    cy.get('[data-testid="news-card"]').should("have.length", 10);
```
Comment (aashitk) [inline cypress/e2e/homepage.cy.ts]: Better if you check length of cards before as well as after load-more button is clicked

**Review by aashitk — CHANGES_REQUESTED** [review]
Added few comments to address changes or provide appropriate reply.
Also can you get rid of `tsconfig 2.json` as well?

File: cypress/e2e/homepage.cy.ts (hunk 1 above)
Comment (MyNewGitHere) [reply cypress/e2e/homepage.cy.ts]: if condition provides a safety check to handle potential scenarios where the intercepted response may not be available

Comment (aashitk) [reply cypress/e2e/homepage.cy.ts]: Tests should always be absolute, avoid conditional testing as much as possible.
If it fails due to potential scenarios, let it fail. That will give us surety our Cypress tests work fine and show failures of API as well.
This makes sure if Cypress works successfully, API is working fine and won't break in real-time execution.

File: cypress/e2e/homepage.cy.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+          .should("eq", initialLength + 5);
+      });
+  });
+
+  it("VerifyNewsCardNavigation", () => {
+    cy.get('[data-testid="news-link"]')
+      .should("have.attr", "target", "_blank")
+      .then(($link) => {
+        const href = $link.attr("href");
+        if (href) {
```
Comment (aashitk) [inline cypress/e2e/homepage.cy.ts]: Remove `if` condition, verify directly.

**Review by aashitk — CHANGES_REQUESTED** [review]
Minor comment

**Review by aashitk — APPROVED** [review]
LGTM

**Pull request merged**